### PR TITLE
Add addon info

### LINKED ISSUE #16: Improve documentation on engine-addon development workflow
I have started out with in-repo engines, which were really easy to setup and get started.
Now, I wanted to use the lazy loading feature, so I migrated them to dedicated addons. However, it turned out that wasn't as straight-forward as I imagined it to be. So I figured the things I "missed" there could also be missed by others, so they might as well go into the README. I'll make a PR with the information I gathered, but wanted to verify a few things so that the information I write there is correct ;)

All dependencies of the host app need to go into the engine's devDependencies
All dependencies of the engine need to go into the engine's dependencies
a.) Everything in the dependencies will be included in the lazy loaded bundle
b.) Nested dependencies do not work correctly. For example, if an engine has a dependency A, which itself has the dependency "ember-d3" (which loads D3.js), then D3.js will not be correctly loaded in the engine. Instead, you need to also include ember-d3 as dependency of your engine.
c.) Remove the ember-disable-prototype-extensions addon from the engine
"ember-engines" should additionally to devDependencies, also go into "peerDependencies"
The engine's dummy app should be setup like the host app

(Moved from https://github.com/ember-engines/ember-engines/issues/307)

### PR DESCRIPTION
**Pull request by mydea**
This adds an "Addon" section with information about how to include addons, addon deduplication and how to manage version conflicts.

Fixes #16 and https://github.com/ember-engines/ember-engines/issues/606

### CODE REVIEW

**Review by rwjblue — APPROVED** [review]
Looks good to me...

**Comment by rwjblue** [comment]
@dgeb - Mind reviewing?

**Comment by villander** [comment]
Super cool @mydea thank you for this work ; )

File: lib/guide/addon/utils/table-of-contents.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ export default [
   { page: 'services', title: 'Sharing Services' },
   { page: 'sharing-components-and-more', title: 'Sharing Components and More' },
   { page: 'lazy-loading', title: 'Lazy Loading' },
+  { page: 'addons', title: 'Engines & Addons' },
```
Comment (villander) [inline lib/guide/addon/utils/table-of-contents.js:11]: `Addons & Dependencies` don't make more sense?

**Comment by dgeb** [comment]
@mydea lots of helpful tips here - thanks so much!

**Pull request merged**